firefox-ios/Client/Frontend/Home/Homepage Rebuild/ContextMenu/ContextMenuState.swift

```diff
@@ -161,25 +178,72 @@ struct ContextMenuState {
                                      iconString: StandardImageIdentifiers.Large.bookmarkSlash,
                                      allowIconScaling: true,
                                      tapHandler: { _ in
-            // TODO: FXIOS-10613 - Add proper actions
+            // TODO: FXIOS-10975 - Add proper actions
         })
     }
 
-    private func getAddBookmarkAction() -> SingleActionViewModel {
+    private func getAddBookmarkAction(site: Site) -> SingleActionViewModel {
         return SingleActionViewModel(title: .BookmarkContextMenuTitle,
                                      iconString: StandardImageIdentifiers.Large.bookmark,
                                      allowIconScaling: true,
                                      tapHandler: { _ in
-            // TODO: FXIOS-10613 - Add proper actions
+            // TODO: FXIOS-10975 - Add proper actions
         })
     }
 
-    private func getShareAction() -> PhotonRowActions {
-        return SingleActionViewModel(title: .ShareContextMenuTitle,
-                                     iconString: StandardImageIdentifiers.Large.share,
-                                     allowIconScaling: true,
-                                     tapHandler: { _ in
-            // TODO: FXIOS-10613 - Add proper actions
-        }).items
+    private func getShareAction(siteURL: String) -> PhotonRowActions {
+        return SingleActionViewModel(
+            title: .ShareContextMenuTitle,
+            iconString: StandardImageIdentifiers.Large.share,
+            allowIconScaling: true,
+            tapHandler: { _ in
+                guard let url = URL(string: siteURL, invalidCharacters: false) else { return }
```

Comment:
Is it weird if we just return here? Do we want to log an error or anything? I am not sure what our criteria is around logging but I can imagine it would be frustrating as a user if they were tapping and nothing was happening

Reply:
thanks for the catch, I will add logs! There are a couple of places that we use this and don't add logs or handle the experience by showing a UI, but I think it's good to at least add a log for now and can revisit in adding error UI for the future new homepage work.